### PR TITLE
Merging rafaelvfalc/vertical-cpu into ufcg-lsd/master

### PR DESCRIPTION
**Pull request by rafaelvfalc**
- Recovering obsolete tests
- Monitor version that support both kubejobs and vertical plugins
- Update of the documentation

### CODE REVIEW

File: monitor/plugins/vertical/plugin.py [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+    def _get_elapsed_time(self):
+        datetime_now = datetime.now()
+        elapsed_time = datetime_now - self.submission_time
+        print "Elapsed Time: %.2f" % elapsed_time.seconds
+
+        return elapsed_time.seconds
+
+    def monitoring_application(self):
+        try:
+            cpu_usage = requests.get('http://%s:5000' % (self.get_api_address())).text
```
Comment (clenimar) [inline monitor/plugins/vertical/plugin.py]: get port number from an attribute instead of hardcoding it.

File: monitor/plugins/vertical/plugin.py [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+        except Exception as ex:
+            print ("Error: No application found for %s. %s remaining attempts"
+                   % (self.app_id, self.attempts))
+
+            print ex.message
+            raise
+
+    def get_cpu_quota(self):
+        try:
+            cpu_quota = requests.get('http://%s:5000/%s' % (self.get_api_address(), self.get_metric_endpoint)).text
```
Comment (clenimar) [inline monitor/plugins/vertical/plugin.py]: ditto (port number).

File: monitor/plugins/vertical/plugin.py [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+            return cpu_quota
+
+        except Exception as ex:
+            print("Error while getting %s metric" % (self.metric_source))
+            print ex.message
+            raise
+
+    def get_api_address(self):
+        # TODO Search a better way to get the internal ip of a node using k8s api
+        nodes_ips = os.popen("kubectl --kubeconfig=%s get nodes -o wide | awk '{print $6}'" % (self.k8s_manifest)).readlines()
```
Comment (clenimar) [inline monitor/plugins/vertical/plugin.py]: use python kubernetes client instead. we should not just assume the user has `kubectl` installed in monitor's host.

File: monitor/utils/influxdb/connector.py [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+        return client
+    
+    def send_metrics(self, measurements):
+
+        measurements = measurements[0]
+        metrics = {}
+
+        metrics['measurement'] = measurements['name']
+        metrics['time'] = datetime.fromtimestamp(measurements['timestamp'] / 1000).strftime('%Y-%m-%dT%H:%M:%SZ')
+        metrics['tags'] = {"host": "server01", "region": "sa-east-1", "job": measurements['dimensions']['application_id']}
```
Comment (clenimar) [inline monitor/utils/influxdb/connector.py]: why are `"host"` and `"region"` hardcoded?

File: docs/plugins/external-api.md [hunk 5]
```diff
@@ -0,0 +1,76 @@
+# External API Plugin
+
+## How does it works?
```
Comment (clenimar) [inline docs/plugins/external-api.md]: typo: s/works/work

File: docs/plugins/external-api.md [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+get_metric_endpoint = cpu-quota
+# Threshold value used to trigger the scalling
+threshold = 0.5
+# The config file that gives permission to access the kubernetes cluster
+k8s_manifest = /home/rafaelvf/.kube/config
+```
+
+## Execute plugin
+
+In order to execute the plugin, a JSON needs to be correctly configurate with all necessary variables that will be used by Asperathos components. Following you can check an example of this JSON file that will be used to sends a POST request to the Asperathos Manager.
```
Comment (clenimar) [inline docs/plugins/external-api.md]: typo: s:/configurate/configured

File: docs/plugins/kubejobs.md [hunk 7]
```diff
@@ -0,0 +1,84 @@
+# KubeJobs Plugin
+
+## How does it works?
```
Comment (clenimar) [inline docs/plugins/kubejobs.md]: ditto (work).

File: docs/plugins/kubejobs.md [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+debug = True
+retries = 5
+
+[kubejobs]
+# No variables are needed
+```
+
+## Execute plugin
+
+In order to execute the plugin, a JSON needs to be correctly configurate with all necessary variables that will be used by Asperathos components. Following you can check an example of this JSON file that will be used to sends a POST request to the Asperathos Manager.
```
Comment (clenimar) [inline docs/plugins/kubejobs.md]: ditto (configured).

**Comment by rafaelvfalc** [comment]
I agree with the change the @clenimar suggest. However, I think it is important to first upload this version into the 'master' branch, create a issue requesting this modification and solve this issue in future PRs. The vertical strategy is still a toy plugin, and should be modificated in the future.

**Comment by clenimar** [comment]
> I agree with the change the @clenimar suggest. However, I think it is important to first upload this version into the 'master' branch, create a issue requesting this modification and solve this issue in future PRs. The vertical strategy is still a toy plugin, and should be modificated in the future.

agreed. as this is a large rebase, we can address those minor issues in follow-up PRs. I'm opening an issue as soon as this one is landed.

**Pull request merged**